src/main/java/com/ibm/cloud/cloudant/kafka/schema/DocumentToMap.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © 2022 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package com.ibm.cloud.cloudant.kafka.schema;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import com.ibm.cloud.cloudant.v1.model.Document;
+
+public class DocumentToMap implements Function<Document, Map<String, Object>> {
```

Comment:
This and `ConnectRecordMapper` both do similar things: implement a function to convert between (connect record/cloudant document) and a `Map`.

- Should they have names which are more aligned? (I'm happy to concede that `ConnectRecordMapper` isn't a great name
- Do they belong in the "schema" namespace, whose name might be a bit misleading?

Reply:
> Should they have names which are more aligned? (I'm happy to concede that ConnectRecordMapper isn't a great name)

Probably. It's a little awkward because the Sink side goes directly from a `SinkRecord` to a `Document`, but the source side goes from `Document`->`Map`->`SourceRecord` so they aren't quite "symmetrical".
I should be able to refactor the `Document` mapper to be a `BiFunction` that takes a `row` and a `topic` and returns a `SourceRecord`.
Then we could be consistent on:
* either `*Record<preposition>Document`
    * `SourceRecordFromDocument`
    * `SinkRecordToDocument`
* or `AToB`
    * `DocumentToSourceRecord`
    * `SinkRecordToDocument`
* or `BFromA`
    * `DocumentFromSinkRecord`
    * `SourceRecordFromDocument`

Any preferences?

> Do they belong in the "schema" namespace, whose name might be a bit misleading?

Probably not, it would be nice to keep them separate from the Kafka Connect implementations though.

Reply:
My order of preference: 1) `AToB` 2) `BFromA` 3) `*Record<preposition>Document`

Reply:
Also `AToB` - only do the refactor to transform to a `SourceRecord` directly if you think it makes sense

Reply:
Alright I'll move the source part here, but I opened https://github.com/IBM/cloudant-kafka-connector/issues/98 for the sink part to try and keep it contained.

Reply:
Done in [abb353b](https://github.com/IBM/cloudant-kafka-connector/pull/93/commits/abb353bc1489222410ff4b8969fa37bd7eee095d)